File: frontend/src/types/job.ts
export type JobStatus = "PENDING" | "PROCESSING" | "COMPLETED" | "CANCELED" | "FAILED" | "UNKNOWN";

export interface JobSummary {
  job_id: number;
  printer: string;
  document: string;
  user: string;
  submitted_at: string | null;
  status: JobStatus;
}

export interface JobDetail extends JobSummary {
  owner: string;
  started_at: string | null;
  completed_at: string | null;
  options: Record<string, string>;
  error: string | null;
}

export interface JobCancelResponse {
  success: boolean;
  job_id: number;
  status: string;
}

File: backend/app/schemas/job.py
from typing import Optional

from pydantic import BaseModel


class JobSummary(BaseModel):
    job_id: int
    printer: str
    document: str
    user: str
    submitted_at: Optional[str] = None
    status: str


class JobDetail(JobSummary):
    owner: str
    started_at: Optional[str] = None
    completed_at: Optional[str] = None
    options: dict[str, str] = {}
    error: Optional[str] = None


class JobCancelResponse(BaseModel):
    success: bool = True
    job_id: int
    status: str = "canceled"
